packages/core/src/helpers/warnDeprecatedProps.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from "react";
+
+type DeprecatedProps = Record<string, string>;
+
+export function useWarnDeprecatedProps<T extends Record<string, any>>(
+  props: T,
+  deprecatedProps: DeprecatedProps,
```

Comment:
maybe alternativeProps? or something similar? it implies for old props, not new one